Zend/zend_exceptions.c

```diff
@@ -184,7 +184,7 @@ static zend_object *zend_default_exception_new_ex(zend_class_entry *class_type,
 	zend_class_entry *base_ce;
 	zend_string *filename;
 
-	Z_OBJ(obj) = object = zend_objects_new(class_type);
+	ZVAL_OBJ(&obj, object = zend_objects_new(class_type));
```

Comment:
The new operation assumes two assignments instead of one. The assignment Z_TYPE_INFO(obj) = IS_OBJECT_EX is going to be useless. There are a lot of similar issues in this patch. Each of them will increase code size and reduce performance.